From 15a70a45336e3a2bd2e27be0b27c61fb2698c5e1 Mon Sep 17 00:00:00 2001
From: Andy Doan <andy@foundries.io>
Date: Wed, 2 Oct 2019 22:53:19 -0500
Subject: [PATCH 16/18] aktualizr-lite: Add reporting of network info

This isn't quite right for anonymous mode but will work in a factory

Signed-off-by: Andy Doan <andy@foundries.io>
---
 src/aktualizr_lite/main.cc                  | 6 ++++++
 src/libaktualizr/primary/sotauptaneclient.h | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/aktualizr_lite/main.cc b/src/aktualizr_lite/main.cc
index 04a5868e..bb6bd1ab 100644
--- a/src/aktualizr_lite/main.cc
+++ b/src/aktualizr_lite/main.cc
@@ -223,6 +223,9 @@ static int daemon_main(LiteClient &client, const bpo::variables_map &variables_m
     if (!client.primary->updateImagesMeta()) {
       LOG_WARNING << "Unable to update latest metadata, using local copy";
     }
+
+    client.primary->reportNetworkInfo();
+
     auto target = find_target(client.primary, hwid, client.config.pacman.tags, "latest");
     if (target != nullptr && !targets_eq(*target, current, compareDockerApps)) {
       LOG_INFO << "Updating base image to: " << *target;
@@ -343,6 +346,9 @@ int main(int argc, char *argv[]) {
 
     Config config(commandline_map);
     config.storage.uptane_metadata_path = BasedPath(config.storage.path / "metadata");
+    if (!config.tls.server.empty()) {
+      config.telemetry.report_network = true;
+    }
     LOG_DEBUG << "Current directory: " << boost::filesystem::current_path().string();
 
     std::string cmd = commandline_map["command"].as<std::string>();
diff --git a/src/libaktualizr/primary/sotauptaneclient.h b/src/libaktualizr/primary/sotauptaneclient.h
index 44dfd463..36b88d5e 100644
--- a/src/libaktualizr/primary/sotauptaneclient.h
+++ b/src/libaktualizr/primary/sotauptaneclient.h
@@ -66,6 +66,7 @@ class SotaUptaneClient {
   bool checkImagesMetaOffline();
   data::InstallationResult PackageInstall(const Uptane::Target &target);
   TargetStatus VerifyTarget(const Uptane::Target &target) { return package_manager_->verifyTarget(target); }
+  void reportNetworkInfo();
 
  protected:
   void addSecondary(const std::shared_ptr<Uptane::SecondaryInterface> &sec);
@@ -113,7 +114,6 @@ class SotaUptaneClient {
   void finalizeAfterReboot();
   void reportHwInfo();
   void reportInstalledPackages();
-  void reportNetworkInfo();
   void verifySecondaries();
   void sendMetadataToEcus(const std::vector<Uptane::Target> &targets);
   std::future<bool> sendFirmwareAsync(Uptane::SecondaryInterface &secondary, const std::shared_ptr<std::string> &data);
-- 
2.23.0

